### ISSUE TITLE
Question about hit_urls / prep

### ISSUE THREAD

**Issue by yamins81** [dicarlolab/mturkutils#7]
@ardila @hahong 

@ardila, in the experiment objects you create, there is a method called hit_urls which basically does three things:
    (1) chunks the trial list (you call this object ExperimentData, which is usually mostly created in the constructor cal into smaller chunks)
    (2) templates and writes out the final HTML files
    (3) uploads them to S3

@hahong similarly, you have a function called prep which:
    (1) creates  the trial list (you call this object 'trials') 
    (2) chunks the trial list
    (3) templates and writes out the final HTML files

In both of your cases, you allow there to be the possibility of multiple ways this prep method could be done.  In @ardila 's case, this is by sometimes overriding the hit_url method.  In @hahong 's case, this is by having a different prep function in each driver script.  

In other words, @hahong  has factored the code so that trial list creation is combined with chunking/templating/writing-out but uploading is separate.   While @ardila has factored the code so that trial list creation is (mostly) separate but chunking and  chunking/templating/writing-out is combined with uploading. 

My question is:  is the any objection to factoring the code as follows:

   (1) a "create_trials" method, which creates a trial_list object, and which various between experiments (and is probably not even implemented in the base class)
    (2) a single standard "prep" method which does chunking/templating/writing-out of final HTML files (both "sandbox" and "production")
    (3) a single standard "uploadHTML" method, which actually uploads the HTMLs.
